### PR TITLE
Fix add jindocache engine for jindodata 5.0 beta release

### PR DESCRIPTION
**Pull request by frankleaf**
### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: .dlv/config.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Configuration file for the delve debugger.
```
Comment (cheyang) [inline .dlv/config.yml:1]: Please remove the file, it's for debugging.

**Review by cheyang — CHANGES_REQUESTED** [review]

File: Library/Caches/go-build/README [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This directory holds cached build artifacts from the Go build system.
```
Comment (cheyang) [inline Library/Caches/go-build/README:1]: Please delete this file.

File: pkg/ddc/jindocache/sync_runtime_test.go [hunk 3]
```diff
@@ -0,0 +1,513 @@
+/*
+Copyright 2022 The Fluid Authors.
```
Comment (cheyang) [inline pkg/ddc/jindocache/sync_runtime_test.go:2]: The header year should be 2023.

File: charts/jindocache/templates/fuse/daemonset.yaml [hunk 4]
```diff
@@ -0,0 +1,247 @@
+{{- $notEnableDnsConfig := (eq .Values.master.dnsServer "1.1.1.1") }}
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ template "jindofs.fullname" . }}-fuse
```
Comment (cheyang) [inline charts/jindocache/templates/fuse/daemonset.yaml:6]: If the chart name is changed to jindocache, I think the name jindofs should be changed to jindocache.

File: charts/jindocache/templates/fuse/daemonset.yaml [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  labels:
+    app: {{ template "jindofs.name" . }}
+    chart: {{ template "jindofs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: jindofs-fuse
```
Comment (cheyang) [inline charts/jindocache/templates/fuse/daemonset.yaml:12]: Should the role name be jindocache-fuse?

**Comment by cheyang** [comment]
@frankleaf please fix the unit test first. thanks.

File: pkg/ddc/factory.go [hunk 6]
```diff
@@ -34,13 +35,14 @@ var buildFuncMap map[string]buildFunc
...
-		"alluxio":  alluxio.Build,
-		"jindo":    jindo.Build,
-		"jindofsx": jindofsx.Build,
-		"goosefs":  goosefs.Build,
-		"juicefs":  juicefs.Build,
-		"thin":     thin.Build,
-		"efc":      efc.Build,
+		"alluxio":    alluxio.Build,
+		"jindo":      jindo.Build,
+		"jindofsx":   jindofsx.Build,
```
Comment (cheyang) [inline pkg/ddc/factory.go:40]: How about puting jindo,jindofsx and jindocache together?

File: .gitignore [hunk 7]
```diff
@@ -40,3 +40,6 @@ sdk
 **/__debug_bin
 **/__pycache__
 
+# local related
```
Comment (cheyang) [inline .gitignore:43]: I think it's not neccessay for developers. Please remove it.

File: pkg/ddc/factory.go (hunk 6 above)
Comment (frankleaf) [reply pkg/ddc/factory.go:40]: done

File: .dlv/config.yml (hunk 1 above)
Comment (frankleaf) [reply .dlv/config.yml]: done

File: Library/Caches/go-build/README (hunk 2 above)
Comment (frankleaf) [reply Library/Caches/go-build/README]: done

File: charts/jindocache/templates/fuse/daemonset.yaml (hunk 4 above)
Comment (frankleaf) [reply charts/jindocache/templates/fuse/daemonset.yaml:6]: done

File: charts/jindocache/templates/fuse/daemonset.yaml (hunk 5 above)
Comment (frankleaf) [reply charts/jindocache/templates/fuse/daemonset.yaml:12]: done

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**